src/SdkCommon/ClientRuntime/ClientRuntime/IRestException.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Rest
+{
+    public interface IRestException
```

Comment:
@dsgouda something to consider this for this feature.
Your intend is to provide a constraint for your generated exceptions and not create exceptions and return for clients to use it.
Move properties out of the Interface and into the base class